### PR TITLE
Add team entity, component dao, and team dao test

### PR DESCRIPTION
**Pull request by matusmacko**

### CODE REVIEW

File: src/main/java/cz/muni/fi/dao/ComponentDao.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.List;
+
+/**
+ * @author Matus Macko
+ */
+public interface ComponentDao {
+    /**
+     * List all available components in DB
+     *
```
Comment (xkureck) [inline src/main/java/cz/muni/fi/dao/ComponentDao.java]: why whitespace here?

Comment (matusmacko) [reply src/main/java/cz/muni/fi/dao/ComponentDao.java]: I checked this on oracle.com and they add a blank line between description and other tags. I guess it is some kind of convention - http://www.oracle.com/technetwork/articles/java/index-137868.html

File: pom.xml [hunk 2]
```diff
@@ -60,7 +60,10 @@
             <artifactId>derby</artifactId>
             <version>10.10.2.0</version>
         </dependency>
-
+        <dependency>
```
Comment (RobertTamas) [inline pom.xml]: I would add dependencies through its own PR, so after merge we can all rebase on it

File: src/test/java/cz/muni/fi/dao/TeamDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ */
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+public class TeamDaoTest extends AbstractTestNGSpringContextTests {
+    @PersistenceContext
+    private EntityManager em;
+
+    @Autowired
+    private TeamDaoImpl teamManager = new TeamDaoImpl();
+
+    private Team team1;
```
Comment (RobertTamas) [inline src/test/java/cz/muni/fi/dao/TeamDaoTest.java]: Maybe use better names for team and car like `Blue team`, Adamek don't like such a names

File: src/test/java/cz/muni/fi/dao/TeamDaoTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+/**
+ * @author Matus Macko
+ */
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+public class TeamDaoTest extends AbstractTestNGSpringContextTests {
```
Comment (RobertTamas) [inline src/test/java/cz/muni/fi/dao/TeamDaoTest.java]: missing test for adding team

File: pom.xml (hunk 2 above)
Comment (matusmacko) [reply pom.xml]: Removed.

File: src/test/java/cz/muni/fi/dao/TeamDaoTest.java (hunk 4 above)
Comment (matusmacko) [reply src/test/java/cz/muni/fi/dao/TeamDaoTest.java]: Added.

**Pull request merged**